From e0792b7605e6d6cb2ebd491025aee7f84d1edbaa Mon Sep 17 00:00:00 2001
Message-ID: <e0792b7605e6d6cb2ebd491025aee7f84d1edbaa.1692864566.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Thu, 17 Feb 2022 22:57:16 +0100
Subject: [PATCH] Avoid writing to system console

Add-ons do not have a working syslog infrastructure. The LOG_CONS
options causes syslog() calls to log to the system console
(/dev/console). Avoid writing to system console as we use logging to
stderr.
---
 src/common/logging.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/logging.cpp b/src/common/logging.cpp
index 5a787e8213..79fccf77ca 100644
--- a/src/common/logging.cpp
+++ b/src/common/logging.cpp
@@ -88,7 +88,7 @@ void otbrLogInit(const char *aProgramName, otbrLogLevel aLevel, bool aPrintStder
     ident = strrchr(aProgramName, '/');
     ident = (ident != nullptr) ? ident + 1 : aProgramName;
 
-    openlog(ident, (LOG_CONS | LOG_PID) | (aPrintStderr ? LOG_PERROR : 0), OTBR_SYSLOG_FACILITY_ID);
+    openlog(ident, LOG_PID | (aPrintStderr ? LOG_PERROR : 0), OTBR_SYSLOG_FACILITY_ID);
     sLevel        = aLevel;
     sDefaultLevel = sLevel;
 }
-- 
2.42.0

